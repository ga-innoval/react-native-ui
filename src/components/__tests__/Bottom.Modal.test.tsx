import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';
import { Text } from 'react-native';
import { BottomModal } from '../Bottom.Modal';

describe('<BottomModal visible={true}/>', () => {
  let component: any = null;
  const mockFn = jest.fn();

  const eventData = {
    nativeEvent: {
      pageX: 10,
      pageY: 10,
    },
  };

  beforeEach(() => {
    component = render(
      <BottomModal onDismiss={mockFn} visible={true}>
        <Text>child component</Text>
      </BottomModal>
    );
  });
  it('should render correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onDissmiss function', () => {
    fireEvent.press(screen.getByTestId('modalPressableArea'), eventData);
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});

describe('<BottomModal visible={false}/>', () => {
  it('should not render', () => {
    render(
      <BottomModal onDismiss={() => {}} visible={false}>
        <Text>Esta es una alerta que puede contener cualquier componente!</Text>
      </BottomModal>
    );
    expect(screen.queryByText(/titulo/i)).toBeNull();
  });
});
