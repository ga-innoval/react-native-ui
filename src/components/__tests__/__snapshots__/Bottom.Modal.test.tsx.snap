// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomModal visible={true} showOpacityMask={false}/> should render correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "height": 667,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "height": 667,
            "position": "relative",
          }
        }
        testID="modalPressableArea"
      />
    </View>
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "white",
          "borderTopLeftRadius": 15,
          "borderTopRightRadius": 15,
          "bottom": 0,
          "flex": 1,
          "position": "relative",
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "paddingHorizontal": 15,
            "paddingVertical": 25,
          }
        }
      >
        <Text>
          child component
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`<BottomModal visible={true}/> should render correctly 1`] = `
[
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "height": 667,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "height": 667,
              "position": "relative",
            }
          }
          testID="modalPressableArea"
        />
      </View>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "white",
            "borderTopLeftRadius": 15,
            "borderTopRightRadius": 15,
            "bottom": 0,
            "flex": 1,
            "position": "relative",
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
              "paddingHorizontal": 15,
              "paddingVertical": 25,
            }
          }
        >
          <Text>
            child component
          </Text>
        </View>
      </View>
    </View>
  </Modal>,
  <View
    style={
      {
        "backgroundColor": "rgba(0,0,0,0.75)",
        "bottom": 0,
        "elevation": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 1,
      }
    }
  />,
]
`;
